### ISSUE TITLE
Audio devices switch automatically when a new device is connected

### ISSUE THREAD

**Issue by dyfer** [essej/sonobus#14]
Experienced on macOS:
When a new device is connected to the system, SonoBus seems to change the device it is using. It seems that the device "index" in the list is maintained (SonoBus keeps using e.g. the 3rd device, even though the list of devices has changed).
IMO the current device should not change when a new device is connected/disconnected.

Moreover, if the currently used device is disconnected, it would be nice if SonoBus notified the user and asked what should be done (e.g. switch devices? stop audio?). The typical issue with the current behavior can happen when the user uses an external sound card and monitors themselves, then (accidentally?) disconnects the soundcard, and SonoBus switches to internal mic / speakers, causing feedback.

**Comment by essej** [comment]
Thanks for reporting this, I hadn't tested when devices were changed... I wouldn't expect this behavior, but I will try to reproduce it.

**Comment by essej** [comment]
OK, looks like the bug is actually in the UI that shows the current audio device... the active audio device in use doesn't *actually* change (as evidenced by the sound engine still working as expected with the already in-use audio device)... the UI is just mistakenly showing the wrong item in the list.

**Comment by dyfer** [comment]
I see now, you are right: indeed it looks like for connecting the new device the only bug is in the UI. This gets confusing if there are multiple devices in the system and the UI gets out of sync with the device that's actually used :)

For disconnecting currently used device, I still think that SonoBus should not change automatically to another device (to avoid the issue I described earlier), but either switch to "no device", or ask the user what to do.

**Comment by essej** [comment]
Yes, your suggestion is a good one if the active device is disconnected
